### ISSUE TITLE
Hunter's pet loyalty level and training level are bugged

### ISSUE THREAD

**Issue by Xkwadraat** [SoloCraftServer/bugtracker#192]
## 🐛 Bug report
The beastmasters pet loyalty level seems not to be gaining as it should when catching experience points. Also its training level stands frozen on -7, it should gain when spending my money on training at the pet trainer. Also it seems that the trained spells don't have affect on the pet (for example: natural armour spell, the resistances spells, etc...)

### Steps to reproduce
1.
2.
3.

**Comment by necropola** [comment]
To me it looks like pets gain loyalty points per "tick" (every 12 seconds) while happy (5x20 = 100 per minute) or at least not unhappy (5x10 = 50 per minute) and bonus loyalty points when killing stuff:
https://github.com/vmangos/core/blob/7e38088bb155c8dd4be588da1d8dfdf4b49252c2/src/game/Objects/Pet.cpp#L839-L868

--> every 12 seconds:
https://github.com/vmangos/core/blob/7e38088bb155c8dd4be588da1d8dfdf4b49252c2/src/game/Objects/Pet.cpp#L754-L760

And the number of loyalty points needed for the next level of loyalty is increasing:
https://github.com/vmangos/core/blob/7e38088bb155c8dd4be588da1d8dfdf4b49252c2/src/game/Objects/Pet.cpp#L36-L55 

This pretty much confirms my own experience.

TL;DR: It takes quite a while for a new pet to become your "Best Friend".
